ingest/change_compactor.go

```diff
@@ -53,17 +57,42 @@ import (
 //     change means the entry exists in a DB.
 //     c. REMOVED it returns error because we can't remove an entry that was
 //     already removed.
+//     d. RESTORED depending on the change compactor's configuration, we may or
+//     may not emit a REMOVE change type for an entry that was restored earlier
+//     in the ledger.
+//  4. If the change is RESTORED it checks if any change related to the given
+//     entry already exists in the cache. If not, it adds the RESTORED change.
+//     Otherwise, it returns an error because only expired entries can be
+//     restored. If the entry was created, updated or removed in the same
+//     ledger, it's not expired.
 type ChangeCompactor struct {
 	// ledger key => Change
 	cache          map[string]Change
 	encodingBuffer *xdr.EncodingBuffer
+	config         *ChangeCompactorConfig
+}
+
+type ChangeCompactorConfig struct {
+	// Determines whether the change compactor emits a REMOVED change when an expired entry
+	// is restored and then removed within the same ledger.
+	// If set to true, a REMOVED change is emitted; if false, REMOVED change is suppressed.
+	EmitExpiredEntryRemovedChange bool
+}
+
+func NewChangeCompactorDefaultConfig() *ChangeCompactorConfig {
+	return &ChangeCompactorConfig{
+		// By default, set it to true to enable the change compactor
+		// to emit REMOVED change for expired entries.
+		EmitExpiredEntryRemovedChange: true,
+	}
 }
 
 // NewChangeCompactor returns a new ChangeCompactor.
-func NewChangeCompactor() *ChangeCompactor {
+func NewChangeCompactor(config *ChangeCompactorConfig) *ChangeCompactor {
```

Comment:
```suggestion
func NewChangeCompactor(config ChangeCompactorConfig) *ChangeCompactor {
```

I think we should pass the config struct by value since it is immutable and not so large